mavparm.py

```diff
@@ -61,7 +61,7 @@ def mavset(self, mav, name, value, retries=3, parm_type=None):
                 if ack is None:
                     time.sleep(0.1)
                     continue
-                if str(name).upper() == str(ack.param_id).upper():
+                if str(name) == ack.param_id.decode('ascii'):
```

Comment:
name probably doesn't need the str(). So I would do `if name.upper() == ack.param_id.decode('ascii', errors='ignore').upper()`

Reply:
This should work in python2 because `'123' == u'123'` produces `True`.